planning/scenario_planning/lane_driving/behavior_planning/lane_change_planner/src/lane_changer.cpp

```diff
@@ -168,22 +181,26 @@ void LaneChanger::init()
   this->get_node_timers_interface()->add_timer(timer_, nullptr);
 }
 
-void LaneChanger::run()
+void LaneChanger::waitForData()
 {
+  using namespace std::literals::chrono_literals;
+  rclcpp::WallRate loop_rate(100ms);
+
   // wait until mandatory data is ready
-  if (!route_handler_ptr_->isHandlerReady()) {
-    RCLCPP_WARN_THROTTLE(
-      get_logger(), *get_clock(), 5000 /* ms */,
-      "waiting for route to be ready");
-    return;
+  while (!route_handler_ptr_->isHandlerReady() && rclcpp::ok()) {
+    RCLCPP_WARN_THROTTLE(get_logger(), *get_clock(), 5000, "waiting for route to be ready");
+    loop_rate.sleep();
   }
-  if (!data_manager_ptr_->isDataReady()) {
+  while (!data_manager_ptr_->isDataReady() && rclcpp::ok()) {
     RCLCPP_WARN_THROTTLE(
       get_logger(),
-      *get_clock(), 5000 /* ms */, "waiting for vehicle pose, vehicle_velocity, and obstacles");
-    return;
+      *get_clock(), 5000, "waiting for vehicle pose, vehicle_velocity, and obstacles");
+    loop_rate.sleep();
   }
```

Comment:
The node will be executed as a single-thread exector so `waitForData()` function will blocked forever since the callbacks wouldn't be called with the `sleep()` function. Unlike ROS1, if we try to call `spin()` functions within rclpp::Node, it will complain about adding the node to an executor twice. When we did the v0.5 porting to ROS2, I've changed the logic so that the data checking will run at every loop instead. There would be some overhead in each loop compared to ROS1 implmentation, but I think it is negligible.

Reply:
Fixed.

Reply:
Fixed.